### PR TITLE
Rewrite `librz/util/bdiff.c`

### PR DESCRIPTION
**Pull request by sandhrabino**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Reimplement `librz/util/bdiff.c` to be license-compatible with the rest of Rizin.

**Test plan**

CI is green

**Closing issues**

Partially addresses https://github.com/rizinorg/rizin/issues/70

### CODE REVIEW

File: librz/include/rz_diff.h [hunk 1]
```diff
@@ -52,6 +52,31 @@ typedef struct rz_diffchar_t {
...
+	int a1;
+	int a2;
+	int b1;
+	int b2;
+} RzHunk;
+
+struct rz_hunklist {
+	struct rz_hunk *base;
+	struct rz_hunk *head;
+} RzHunkList;
```
Comment (karliss) [inline librz/include/rz_diff.h:78]: You can't simply copy paste parts of GPL licensed code into rest of the codebase and rename it to RZ. That defeats the whole point of #70.

Comment (sandhrabino) [reply librz/include/rz_diff.h:78]: I'm sorry, I will rewrite the code and make another commit.

**Comment by sandhrabino** [comment]
> Just remove it.

I will remove it.

**Comment by XVilka** [comment]
These format changes are wrong - just look at the list of the affected unrelated files.

**Comment by sandhrabino** [comment]
> These format changes are wrong - just look at the list of the affected unrelated files.

Yes, I am aware, I have made the changes locally, will push them with a few other changes.

**Comment by wargio** [comment]
which part of the GPL stuff are you rewriting?

**Comment by sandhrabino** [comment]
> which part of the GPL stuff are you rewriting?

I am rewriting bdiff.c

**Comment by XVilka** [comment]
@sandrabeme are there any updates on this?
Please also rebase on top of the latest dev.

**Comment by sandhrabino** [comment]
> @sandrabeme are there any updates on this?
> Please also rebase on top of the latest dev.

I have not made much progress here.

**Pull request closed without merging**